### PR TITLE
feat: dashboard variable chaining

### PR DESCRIPTION
**Pull request by srikanthccv**
Part of #1973 

This adds the ability to use the variables inside a query in dashboards. This PR only contains the query service part because it can go without waiting for frontend work.

### CODE REVIEW

File: pkg/query-service/app/http_handler.go [hunk 1]
```diff
@@ -793,6 +794,54 @@ func (aH *APIHandler) queryDashboardVars(w http.ResponseWriter, r *http.Request)
...
+	if err := json.NewDecoder(r.Body).Decode(&postData); err != nil {
+		return "", fmt.Errorf("failed to decode request body: %v", err)
+	}
+
+	query := strings.TrimSpace(postData.Query)
+
+	if query == "" {
+		return "", fmt.Errorf("query is required")
+	}
+	if strings.Contains(strings.ToLower(query), "alter table") {
```
Comment (nityanandagohain) [inline pkg/query-service/app/http_handler.go:809]: Should we also check from truncate and drop table ?

Comment (srikanthccv) [reply pkg/query-service/app/http_handler.go:809]: Sure, let me add all the mutation kinds here.

Comment (nityanandagohain) [reply pkg/query-service/app/http_handler.go:809]: Yeah, or can create a separate file for checking this, can be utilised by other functions as well.

Comment (srikanthccv) [reply pkg/query-service/app/http_handler.go:809]: This is a very basic check against the queries run. I am not sure if it is worth being shared. Ideally, the user should be limited to read privileges here. I believe I will keep limited to this for now and create a new issue for read only user.

**Review by nityanandagohain — APPROVED** [review]

**Pull request merged**